tests/protocols/test_http.py

```diff
@@ -700,3 +704,18 @@ def test_scopes(asgi2or3_app, expected_scopes, protocol_cls):
     protocol.data_received(SIMPLE_GET_REQUEST)
     protocol.loop.run_one()
     assert expected_scopes == protocol.scope.get("asgi")
+
+
+@pytest.mark.parametrize("protocol_cls", HTTP_PROTOCOLS)
+def test_invalid_http_request(protocol_cls, caplog):
+    def app(scope):
+        return Response("Hello, world", media_type="text/plain")
+
+    caplog.set_level(logging.INFO, logger="uvicorn.error")
+    logging.getLogger("uvicorn.error").propagate = True
+
+    protocol = get_connected_protocol(app, protocol_cls)
+    protocol.data_received(INVALID_REQUEST)
+    assert b"" in protocol.transport.buffer
+    assert b"" in protocol.transport.buffer
```

Comment:
Not sure what's the purpose of these two lines? Did you mean…?

```suggestion
    assert not protocol.transport.buffer
```